### PR TITLE
[Nvidia] Remove SAI_HOSTIF_OPER_STATUS_UPDATE_BY_APP, set it only for certain SKUs

### PR DESCRIPTION
**Pull request by stepanblyschak**
#### Why I did it

We found some race condition issues when using SAI_HOSTIF_OPER_STATUS_UPDATE_BY_APP affecting warm-reboot flow.

##### Work item tracking
- Microsoft ADO **(number only)**:

#### How I did it

Unset SAI_HOSTIF_OPER_STATUS_UPDATE_BY_APP for all SKUs except those which need this functionality.

#### How to verify it

Build and run.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [ ] 202205
- [ ] 202211
- [ ] 202305
- [x] 202311

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Review by volodymyrsamotiy — APPROVED** [review]

File: device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-O128/sai.profile [hunk 1]
```diff
@@ -1 +1,2 @@
 SAI_INIT_CONFIG_FILE=/usr/share/sonic/hwsku/sai_sn5600_128x400g_1x25g.xml
```
Comment (kperumalbfn) [inline device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-O128/sai.profile:1]: How does this work for other SKUs?

Comment (stepanblyschak) [reply device/mellanox/x86_64-nvidia_sn5600-r0/Mellanox-SN5600-O128/sai.profile:1]: We'd like to keep this flag for this specific SKU as there's a specific requirement to have this feature enabled there

**Review by liat-grozovik — APPROVED** [review]

**Comment by liat-grozovik** [comment]
@yxieca  please help to resolve ms_conflict and unblock the PR from getting merged.

**Pull request merged**